tests/routes/test_vespasearch.py

```diff
@@ -0,0 +1,743 @@
+        body_with_filters = _make_search_request(client, params=params)
+        filtered_family_slugs = [
+            f["family_slug"] for f in body_with_filters["families"]
+        ]
+
+        assert family["family_slug"] in filtered_family_slugs
+
+
+@pytest.mark.search
+@pytest.mark.parametrize("label,query", [("search", "the"), ("browse", "")])
+def test_keyword_region_filters(label, query, test_vespa, client, test_db, monkeypatch):
+    monkeypatch.setattr(search, "_VESPA_CONNECTION", test_vespa)
+    _populate_search_db_families(test_db)
+    base_params = {"query_string": query}
+
+    # Get regions of all documents and iterate over them
+    # to confirm the originals are returned when filtered on
+    all_body = _make_search_request(client, params=base_params)
+    families = [f for f in all_body["families"]]
+    assert len(families) >= 4
+
+    for family in families:
+        country_code = family["family_geography"]
+
+        # Fixture for UNFCCC.non-party.1267.0 has a non geography (XAA)
+        if country_code == "Other":
+            return
+
+        parent_id = (
+            test_db.query(Geography)
+            .filter(Geography.value == country_code)
+            .first()
+            .parent_id
+        )
+        region = test_db.query(Geography).filter(Geography.id == parent_id).first()
+
+        params = {**base_params, **{"keyword_filters": {"regions": [region.slug]}}}
+        body_with_filters = _make_search_request(client, params=params)
+        filtered_family_slugs = [
+            f["family_slug"] for f in body_with_filters["families"]
+        ]
+        assert family["family_slug"] in filtered_family_slugs
+
+
+@pytest.mark.search
+@pytest.mark.parametrize("label,query", [("search", "the"), ("browse", "")])
+def test_invalid_keyword_filters(
+    label, query, test_vespa, test_db, monkeypatch, client
+):
+    monkeypatch.setattr(search, "_VESPA_CONNECTION", test_vespa)
+    _populate_search_db_families(test_db)
+
+    response = client.post(
+        SEARCH_ENDPOINT,
+        json={
+            "query_string": query,
+            "keyword_filters": {
+                "geographies": ["kenya"],
+                "unknown_filter_no1": ["BOOM"],
+            },
+        },
+    )
+    assert response.status_code == 422
+
+
+@pytest.mark.search
+@pytest.mark.parametrize(
+    "year_range", [(None, None), (1900, None), (None, 2020), (1900, 2020)]
+)
+def test_year_range_filterered_in(year_range, test_vespa, test_db, monkeypatch, client):
+    monkeypatch.setattr(search, "_VESPA_CONNECTION", test_vespa)
+    _populate_search_db_families(test_db)
+
+    # Search
+    params = {"query_string": "and", "year_range": year_range}
+    body = _make_search_request(client, params=params)
+    assert len(body["families"]) > 0
+
+    # Browse
+    params = {"query_string": "", "year_range": year_range}
+    body = _make_search_request(client, params=params)
+    assert len(body["families"]) > 0
```

Comment:
Do we need to interrogate the family events to confirm the year?

Reply:
I'm going to ticket this along with your other comment, it's a good shout but I'm thinking not one for today